src/enclave/thread_local.cpp

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#include <cstdint>
+#include <map>
+#include <thread>
+
+std::map<std::thread::id, uint16_t> thread_ids;
```

Comment:
Does this have to be in its own file?

Reply:
It's probably not a bad start to a libccf.a or something like that?

Reply:
This map is populated in `enclave/main.cpp`, and we already do the rest of our enclave static initialisation there, so I think this should move there as well.

Reply:
This definition is needed is a bunch of places so that .a is more appropriate. However, creating libccf.a should be a separate work item.

Reply:
Created #716